snapcraft/internal/lifecycle.py

```diff
@@ -109,12 +109,13 @@ def execute(step, project_options, part_names=None):
     """
     config = snapcraft.internal.load_config(project_options)
     repo.Repo.install_build_packages(config.build_tools)
+    build_packages = repo.Repo.get_installed_package_list(config.build_tools)
 
     if (os.environ.get('SNAPCRAFT_SETUP_CORE') and
             config.data['confinement'] == 'classic'):
         _setup_core(project_options.deb_arch)
 
-    _Executor(config, project_options).run(step, part_names)
+    _Executor(config, project_options, build_packages).run(step, part_names)
```

Comment:
Now that we actually have a function for extracting build package versions using the repo and config (which is already passed to the executor), we can probably stop passing them around and calculate them when needed. When is the latest we could do it? Does it even need to be in the lifecycle?